x-pack/plugins/cases/server/services/attachments/operations/get.test.ts

```diff
@@ -25,6 +38,163 @@ describe('AttachmentService getter', () => {
     });
   });
 
+  describe('bulkGet', () => {
+    describe('Decoding', () => {
+      it('does not throw when the response has the required fields', async () => {
+        unsecuredSavedObjectsClient.bulkGet.mockResolvedValue({
+          saved_objects: [createUserAttachment()],
+        });
+
+        await expect(attachmentGetter.bulkGet(['1'])).resolves.not.toThrow();
+      });
+
+      it('does not modified the error saved objects', async () => {
+        unsecuredSavedObjectsClient.bulkGet.mockResolvedValue({
+          saved_objects: [createUserAttachment(), createErrorSO()],
+        });
+
+        const res = await attachmentGetter.bulkGet(['1', '2']);
+
+        expect(res).toStrictEqual({ saved_objects: [createUserAttachment(), createErrorSO()] });
+      });
+
+      it('strips excess fields', async () => {
+        unsecuredSavedObjectsClient.bulkGet.mockResolvedValue({
+          saved_objects: [{ ...createUserAttachment({ foo: 'bar' }) }],
+        });
+
+        const res = await attachmentGetter.bulkGet(['1']);
+        expect(res).toStrictEqual({ saved_objects: [createUserAttachment()] });
+      });
+
+      it('throws when the response is missing the attributes.comment field', async () => {
+        const invalidAttachment = createUserAttachment();
+        unset(invalidAttachment, 'attributes.comment');
+
+        unsecuredSavedObjectsClient.bulkGet.mockResolvedValue({
+          saved_objects: [invalidAttachment],
+        });
+
+        await expect(attachmentGetter.bulkGet(['1'])).rejects.toThrowErrorMatchingInlineSnapshot(
+          `"Invalid value \\"undefined\\" supplied to \\"comment\\",Invalid value \\"user\\" supplied to \\"type\\",Invalid value \\"undefined\\" supplied to \\"alertId\\",Invalid value \\"undefined\\" supplied to \\"index\\",Invalid value \\"undefined\\" supplied to \\"rule\\",Invalid value \\"undefined\\" supplied to \\"actions\\",Invalid value \\"undefined\\" supplied to \\"externalReferenceAttachmentTypeId\\",Invalid value \\"undefined\\" supplied to \\"externalReferenceMetadata\\",Invalid value \\"undefined\\" supplied to \\"externalReferenceId\\",Invalid value \\"undefined\\" supplied to \\"externalReferenceStorage\\",Invalid value \\"undefined\\" supplied to \\"persistableStateAttachmentTypeId\\",Invalid value \\"undefined\\" supplied to \\"persistableStateAttachmentState\\""`
```

Comment:
I would expect a message only about the `comment`. Is it because io-ts tries to find a matching definion on the other attachment types?

Reply:
Yeah that's what I thought originally too. Yeah I think the issue is that io-ts doesn't know what type to use from the union because it doesn't match any of them.